docs/reference/query-dsl/weighted-tokens-query.asciidoc

```diff
@@ -0,0 +1,122 @@
+[[query-dsl-weighted-tokens-query]]
+=== Weighted tokens query
+++++
+<titleabbrev>Weighted tokens</titleabbrev>
+++++
+
+preview::[]
+
+The weighted tokens query requires a list of token-weight pairs that are sent in with a query rather than calculated using a {nlp} model.
+These token pairs are then used in a query against a <<sparse-vector,sparse vector>> or <<rank-features,rank features>> field.
+
+Weighted tokens queries are useful when you want to use an external query expansion model, or quickly prototype changes without reindexing a new model.
+
+[discrete]
+[[weighted-tokens-query-ex-request]]
+==== Example request
+
+[source,console]
+----
+POST _search
+{
+  "query": {
+    "weighted_tokens": {
+      "query_expansion_field": {
+        "tokens": {"2161": 0.4679, "2621": 0.307, "2782": 0.1299, "2851": 0.1056, "3088": 0.3041, "3376": 0.1038, "3467": 0.4873, "3684": 0.8958, "4380": 0.334, "4542": 0.4636, "4633": 2.2805, "4785": 1.2628, "4860": 1.0655, "5133": 1.0709, "7139": 1.0016, "7224": 0.2486, "7387": 0.0985, "7394": 0.0542, "8915": 0.369, "9156": 2.8947, "10505": 0.2771, "11464": 0.3996, "13525": 0.0088, "14178": 0.8161, "16893": 0.1376, "17851": 1.5348, "19939": 0.6012},
+        "pruning_config": {
+          "tokens_freq_ratio_threshold": 5,
+          "tokens_weight_threshold": 0.4,
+          "only_score_pruned_tokens": false
+        }
```

Comment:
In this example, there is a token-weight pair ("7387": 0.0985). My understanding is that whenever a client sends a search query using **weighted_tokens**, the Elasticsearch will ignore all the documents which have "7387" present because "tokens_weight_threshold" is 0.4. Am I right?